visualization/app/codeCharta/state/store/dynamicSettings/searchedNodePaths/searchedNodePaths.reducer.ts

```diff
@@ -1,9 +1,9 @@
 import { SearchedNodePathsAction, SearchedNodePathsActions, setSearchedNodePaths } from "./searchedNodePaths.actions"
 
-export function searchedNodePaths(state: string[] = setSearchedNodePaths().payload, action: SearchedNodePathsAction): string[] {
+export function searchedNodePaths(state: Set<string> = setSearchedNodePaths().payload, action: SearchedNodePathsAction): Set<string> {
 	switch (action.type) {
 		case SearchedNodePathsActions.SET_SEARCHED_NODE_PATHS:
-			return [...action.payload]
+			return action.payload
```

Comment:
Is this just passing through the reference? Should we clone it?

Reply:
cloning a Set with rfdc is buggy.
But we could do something like 
```
return new Set(Array.from(action.payload))
```